File: backend-v2/app/models/car.rb
# app/models/car.rb
class Car < ApplicationRecord
  validates :owner, presence: true, length: { maximum: 40 }
  validates :branch, presence: true, length: { maximum: 30 }
  validates :model, presence: true, length: { maximum: 30 }
  validates :color, presence: true, length: { maximum: 12 }

  validates :frabricated,
            presence: true,
            numericality: {
              only_integer: true,
              greater_than: 1900,
              less_than_or_equal_to: Date.current.year + 1
            }

  validates :plate,
            presence: true,
            length: { maximum: 7 },
            uniqueness: true

  has_many :technical_reviews, dependent: :destroy
  has_many :infractions, dependent: :destroy
  has_many :complains, dependent: :destroy
end

File: road_eye/lib/models/car.dart
// lib/models/car.dart

class Car {
  final int? id;
  final String owner;
  final String branch;
  final String model;
  final String color;
  final int fabricated;
  final String plate;

  Car({
    this.id,
    required this.owner,
    required this.branch,
    required this.model,
    required this.color,
    required this.fabricated,
    required this.plate,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'] as int?,
      owner: json['owner'] as String,
      branch: json['branch'] as String,
      model: json['model'] as String,
      color: json['color'] as String,
      fabricated: json['frabricated'] as int, 
      plate: json['plate'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {
      'owner': owner,
      'branch': branch,
      'model': model,
      'color': color,
      'frabricated': fabricated,
      'plate': plate,
    };
    if (id != null) {
      data['id'] = id;
    }
    return data;
  }

  @override
  String toString() {
    return 'Car(id: $id, owner: $owner, branch: $branch, model: $model, color: $color, fabricated: $fabricated, plate: $plate)';
  }
}